src/client/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,67 @@
+import './Autocomplete.scss'
+import React from 'react'
+
+import axios from 'axios'
+import Turnstone from 'turnstone'
+
+type Props = {
+  // styles: Record of autocomplete class names
+  styles?: Record<string, string>
+  // listbox: Object containing autocomplete options,
+  // mainly used for selecting correct key for displayName for object
+  listbox: Record<string, unknown>
+  // url: Should support query, limit params
+  url: string
+  disabled: boolean
+
+  value: string
+  name?: string
+  onChange: (value: any) => void
+}
```

Comment:
Should be dynamic enough for our use case(s).
If more complexity is needed in future cases, we can alter the component